src/test/api_compatibility.ts

```diff
@@ -0,0 +1,93 @@
+
+import { assert, expect } from 'chai';
+import { BinaryCarrier, FORMAT_BINARY, FORMAT_TEXT_MAP, Reference, REFERENCE_CHILD_OF, Span, Tracer } from '../index';
+
+export interface ApiCompatibilityChecksOptions {
+    /** a boolean that controls whether or not to verify baggage values */
+    checkBaggageValues?: boolean;
+}
+
+/**
+ * A function that takes a tracer factory, and tests wheter the initialized tracer
+ * fulfills Opentracing's api requirements.
+ *
+ * @param {object} createTracer - a factory function that allocates a tracer.
+ * @param {object} [options] - the options to be set on api compatibility
+ */
+export function apiCompatibilityChecks(createTracer = () => new Tracer(), options: ApiCompatibilityChecksOptions = {}): void {
+
+    describe('OpenTracing API Compatibility', () => {
+        let tracer: Tracer;
+        let span: Span;
+
+        beforeEach(() => {
+            tracer = createTracer();
+            span = tracer.startSpan('test-span');
+        });
+
+        describe('Tracer', () => {
+
+            describe('startSpan', () => {
+                it('should handle Spans and SpanContexts', () => {
+                    expect(() => { tracer.startSpan('child', { childOf : span }); }).to.not.throw(Error);
+                    expect(() => { tracer.startSpan('child', { childOf : span.context() }); }).to.not.throw(Error);
```

Comment:
It can be useful for pure JS implementations. It checks that a given implementation satisfies the API, and in some cases checks the behavior, like chaining of methods, and baggage.